src/compiler/sourcemap.ts

```diff
@@ -4,7 +4,11 @@ namespace ts {
         extendedDiagnostics?: boolean;
     }
 
-    export function createSourceMapGenerator(host: EmitHost, file: string, sourceRoot: string, sourcesDirectoryPath: string, generatorOptions: SourceMapGeneratorOptions): SourceMapGenerator {
+    declare let TextDecoder: new() => { decode(buffer: ArrayBuffer | ArrayBufferView): string };
+    TextDecoder ||= require("util").TextDecoder

```

Comment:
It's not safe to reference an undefined global.
```suggestion
const decoder = new (typeof TextDecoder !== 'undefined' ? TextDecoder : require('util').TextDecoder)()